R/bambu-processReads_scoreReadClasses.R

```diff
@@ -0,0 +1,215 @@
+                          numTend = polyATerminals$numTend)
+  return(rowData)
+}
+
+#' % of a genes read counts assigned to each read class
+calculateGeneProportion = function(counts, geneIds){
+  countsTBL <- tibble(counts, geneIds) %>%
+    group_by(geneIds) %>% mutate(geneReadProp = counts/sum(counts), 
+                                 geneReadCount = sum(counts))
+  return(countsTBL)
+
+}
+
+#' returns number of ref anno each read class is a subset of
+isReadClassCompatible =  function(query, subject){
+  outData <- data.frame(compatible=rep(0, length(query)), equal = rep(FALSE, length(query)))
+  query <- cutStartEndFromGrangesList(query)
+  subject <- cutStartEndFromGrangesList(subject)
+  
+  olap = findOverlaps(query, subject, ignore.strand = F, type = 'within')
+  query <- query[queryHits(olap)]
+  subject <- subject[subjectHits(olap)]
+  splice <- myGaps(query)
+
+  comp <- myCompatibleTranscription(query = query, subject = subject, splice = splice)
+  equal <- elementNROWS(query)==elementNROWS(subject) & comp
+
+  counts <- countQueryHits(olap[comp])
+
+  outData$compatible <- counts
+  outData$equal <- countQueryHits(olap[equal])>0
+  
+  return(outData)
+}
+
+#' returns number of A/T's each read class aligned 5' and 3' end
+countPolyATerminals = function(grl, genomeSequence){
+  start <- resize(granges(unlist(selectStartExonsFromGrangesList(grl, exonNumber = 1), 
+                                 use.names = F)), width = 10, fix = 'start', ignore.strand=F)
+  end <- resize(granges(unlist(selectEndExonsFromGrangesList(grl, exonNumber = 1), 
+                               use.names = F)), width = 10, fix = 'end', ignore.strand=F)
+  startTemp = start
+  start[which(unlist(unique(strand(grl))) == '-')] = end[which(unlist(unique(strand(grl))) == '-')]
+  end[which(unlist(unique(strand(grl))) == '-')] = startTemp[which(unlist(unique(strand(grl))) == '-')]
+  startSeqs = BSgenome::getSeq(genomeSequence,start)
+  endSeqs = BSgenome::getSeq(genomeSequence,end)
+  numATstart = letterFrequency(startSeqs, c("A","T"))
+  numATend= letterFrequency(endSeqs, c("A","T"))
+  return(data.frame(numAstart=numATstart[,"A"], numAend= numATend[,"A"],  
+                    numTstart=numATstart[,"T"], numTend=numATend[,"T"]))
+}
+
+#' calculates a score based on how likely the read class is associated with a 
+#' real gene
+getGeneScore = function(se){
+  geneFeatures = prepareGeneModelFeatures(rowData(se))
+  features = cbind(geneFeatures$numReads, geneFeatures$strand_bias, 
+                   geneFeatures$numRCs, geneFeatures$numExons, 
+                   geneFeatures$isSpliced , geneFeatures$highConfidence)
+  if(checkFeatures(geneFeatures)){
+    geneModel = fit_xgb(features,geneFeatures$labels)
+    geneScore = as.numeric(predict(geneModel, as.matrix(features), 
+                                   s = "lambda.min",type="response"))
+    
+    geneFDR = calculateFDR(geneScore, labels)
```

Comment:
might not work, labels is not defined

Reply:
https://github.com/GoekeLab/bambu/pull/200#discussion_r600297844 discussed here